rust/lance/src/dataset/scanner.rs

```diff
@@ -3402,19 +3562,20 @@ mod test {
                     .filter("i > 10")?
                     .prefilter(true))
             },
-            "Projection: fields=[i, s, vec, _distance]
-  Take: columns=\"_distance, _rowid, vec, i, s\"
-    KNNFlat: k=5 metric=l2
-      RepartitionExec: partitioning=RoundRobinBatch(1), input_partitions=2
-        UnionExec
-          Projection: fields=[_distance, _rowid, vec]
-            KNNFlat: k=5 metric=l2
-              FilterExec: i@1 > 10
-                LanceScan: uri=..., projection=[vec, i], row_id=true, ordered=false
-          Take: columns=\"_distance, _rowid, vec\"
-            SortExec: TopK(fetch=5), expr=...
-              KNNIndex: name=..., k=5, deltas=1
-                ScalarIndexQuery: query=i > 10",
+            "ProjectionExec: expr=[i@0 as i, s@1 as s, vec@2 as vec, _distance@3 as _distance]
+  Projection: fields=[i, s, vec, _distance]
```

Comment:
It would be nice to not have this kind of no-op projection in the final plan. Could be another optimizer rule later: https://github.com/lancedb/lance/blob/main/rust/lance/src/io/exec/optimizer.rs